src/util/locking/RedisResourceLocker.ts

```diff
@@ -0,0 +1,169 @@
+import { assert } from 'console';
+import type { RedisClient } from 'redis';
+import { createClient } from 'redis';
+import type { Lock } from 'redlock';
+import Redlock from 'redlock';
+import type { ResourceIdentifier } from '../../ldp/representation/ResourceIdentifier';
+import { getLoggerFor } from '../../logging/LogUtil';
+import { InternalServerError } from '../errors/InternalServerError';
+import type { ResourceLocker } from './ResourceLocker';
+
+// The ttl set on a lock, not really important cause redlock wil not handle expiration
+const ttl = 10000;
+// The default redlock config
+const defaultRedlockConfig = {
+  // The expected clock drift; for more details
+  // see http://redis.io/topics/distlock
+  // Multiplied by lock ttl to determine drift time
+  driftFactor: 0.01,
+  // The max number of times Redlock will attempt
+  // to lock a resource before erroring
+  retryCount: 1000000,
+  // The time in ms between attempts
+  retryDelay: 200,
+  // The max time in ms randomly added to retries
+  // to improve performance under high contention
+  // see https://www.awsarchitectureblog.com/2015/03/backoff.html
+  retryJitter: 200,
+};
+/**
+ * A locking system that uses a Redis server or any number of
+ * Redis nodes / clusters
+ * This solution has issues though:
+ *  - Redlock wants to handle expiration itself, this is against the design of a ResourceLocker.
+ *    The workaround for this is to extend an active lock indefinitely.
+ *  - This solution is not multithreaded!
+ *  - Redlock does not provide the ability to see which locks have expired
+ */
+export class RedisResourceLocker implements ResourceLocker {
+  protected readonly logger = getLoggerFor(this);
+
+  public redlock: Redlock;
```

Comment:
Instead of having this public so its functions can be overwritten in the unit tests (at least I'm guessing that's why this is public?), this should be private and the constructor should be mocked in the unit tests. In your unit tests you would then need `jest.mock('redlock')` at the top level of the file, see https://jestjs.io/docs/es6-class-mocks#replacing-the-mock-using-mockimplementation-or-mockimplementationonce . Or you could make a mockfile to mock the functionality as well if you prefer that (also in the same link). We already have one such mock in the `__mocks__` folder.